lib/AST/ASTImporter.cpp

```diff
@@ -2841,16 +2905,19 @@ Decl *ASTNodeImporter::VisitIndirectFieldDecl(IndirectFieldDecl *D) {
     NamedChain[i++] = cast<NamedDecl>(D);
   }
 
-  IndirectFieldDecl *ToIndirectField = IndirectFieldDecl::Create(
-      Importer.getToContext(), DC, Loc, Name.getAsIdentifierInfo(), T,
-      {NamedChain, D->getChainingSize()});
+  llvm::MutableArrayRef<NamedDecl *> CH = {NamedChain, D->getChainingSize()};
+  bool AlreadyImported;
+  IndirectFieldDecl *ToIndirectField;
+  std::tie(ToIndirectField, AlreadyImported) = CreateDecl<IndirectFieldDecl>(
+      D, Importer.getToContext(), DC, Loc, Name.getAsIdentifierInfo(), T, CH);
+  if (AlreadyImported)
+    return ToIndirectField;
```

Comment:
Is it no problem that NamedChain is created with `new` and not deleted if AlreadyImported is true? (But the array is even not deleted by IndirectFieldDecl.)

Reply:
Well, we will leak memory in this case.  Note, we could deallocate `NamedChain` , but I am not sure if that deallocation is safe.
We could try synthesize a test where `IndirectFieldDecl` is already imported and we could check with ASAN whether the deallocation is ok.
However, I am not sure if we could come up with a test where there is an already imported node from this kind. Perhaps a FIXME note is best here.

Reply:
Even better idea: We put a warning here when this case happens and hopefully we can reduce later if we find the warning in real projects.